src/queue.zig

```diff
@@ -3,126 +3,220 @@ const assert = std.debug.assert;
 
 const constants = @import("./constants.zig");
 
+const QueueLink = extern struct {
+    next: ?*QueueLink = null,
+};
+
 /// An intrusive first in/first out linked list.
 /// The element type T must have a field called "next" of type ?*T
```

Comment:
```suggestion
/// The element type T must have a field called "link" of type `QueueLink`.
```

This same comment is repeated in `stack.zig` as well.